components/email/src/Email.svelte

```diff
@@ -193,6 +204,51 @@
     );
   }
 
+  //#region Clean Conversation
+  // If a user sets message_body_type to "clean", expand their message to clean conversation.
+  // This requires them to have access to the Nylas Neural API.
+
+  const CONVERSATION_ENDPOINT_MAX_MESSAGES = 20;
+
+  function cleanConversation() {
+    if (activeThread) {
+      fetchCleanConversations({
+        component_id: id,
+        message_id: activeThread.messages
+          .slice(-CONVERSATION_ENDPOINT_MAX_MESSAGES)
+          .map((message) => message.id),
+      }).then((results) => {
+        results.forEach((msg: Message) => {
+          let existingMessage = activeThread.messages.find(
+            (message) => message.id === msg.id,
+          );
+          if (existingMessage) {
+            existingMessage.conversation = msg.conversation;
+          }
+        });
+        activeThread.messages = activeThread.messages;
+      });
+    } else if (message) {
+      fetchCleanConversations({
+        component_id: id,
+        message_id: [message.id],
+      }).then((results) => {
+        results.forEach((msg: Message) => {
+          message.conversation = msg.conversation;
+        });
+        activeThread.messages = activeThread.messages;
+      });
+    }
+  }
```

Comment:
You'll notice we're doing some `if thread, else if message` stuff here (and elsewhere in the component).

This is probably less DRY than it should be, and we ought to spend some time reducing complexity here in a future PR.

Reply:
Agreed! I'll create a story for refactoring Email component in wonka-next 👍🏽